src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/OffenderService.kt

```diff
@@ -18,52 +18,67 @@ class OffenderService(
   private val client: ProbationOffenderSearchClient,
   private val featureFlag: FeatureFlag,
 ) {
-
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
+    const val NO_RECORDS_MESSAGE = "No Delius matching records exist"
+    const val EXACT_MATCH_MESSAGE = "Exact delius match found - adding offender to person record"
+    const val PARTIAL_MATCH_MESSAGE = "Partial Delius match found"
+    const val MULTIPLE_MATCHES_MESSAGE = "Multiple Delius matches found"
   }
 
   fun processAssociatedOffenders(personEntity: PersonEntity, person: Person) {
     log.debug("Entered processAssociatedOffenders")
     if (featureFlag.isDeliusSearchEnabled()) {
       val offenderDetails = client.getOffenderDetail(SearchDto.from(person))
-      offenderDetails?.let {
-        if (matchesExistingOffenderExactly(it, person)) {
-          log.debug("Exact delius match found - adding offender to person record")
-          personRecordService.addOffenderToPerson(personEntity, Person.from(it[0]))
-          telemetryService.trackEvent(
-            TelemetryEventType.DELIUS_MATCH_FOUND,
-            mapOf(
-              "UUID" to personEntity.personId.toString(),
-              "PNC" to person.otherIdentifiers?.pncIdentifier?.pncId,
-              "CRN" to person.otherIdentifiers?.crn,
-            ),
-          )
-        } else if (matchesExistingOffenderPartially(it, person)) {
-          log.debug("Partial Delius match found")
-          telemetryService.trackEvent(
-            TelemetryEventType.DELIUS_PARTIAL_MATCH_FOUND,
-            mapOf(
-              "UUID" to personEntity.personId.toString(),
-              "PNC" to person.otherIdentifiers?.pncIdentifier?.pncId,
-              "CRN" to person.otherIdentifiers?.crn,
-            ),
-          )
-        }
-      }
+
       if (offenderDetails.isNullOrEmpty()) {
-        log.debug("No Delius matching records exist")
-        telemetryService.trackEvent(
-          TelemetryEventType.DELIUS_NO_MATCH_FOUND,
-          mapOf("UUID" to personEntity.personId.toString(), "PNC" to person.otherIdentifiers?.pncIdentifier?.pncId),
-        )
+        logAndTrackEvent(NO_RECORDS_MESSAGE, TelemetryEventType.DELIUS_NO_MATCH_FOUND, personEntity, person)
+      } else {
+        when {
```

Comment:
Nice - like this way of using `when`

What happens if none of these conditions are met?